src/python/turicreate/test/test_sarray.py

```diff
@@ -73,26 +73,38 @@ def __test_almost_equal(self, _sarray, _data, _type):
             else:
                 self.assertAlmostEqual(l[i], _data[i])
 
-    def __test_creation(self, data, dtype, expected):
```

Comment:
split this helper is due to the fact that generators, map, and filter can be used in only one pass.